java/gazelle/generate.go

```diff
@@ -346,6 +348,36 @@ func (l javaLang) GenerateRules(args language.GenerateArgs) language.GenerateRes
 	return res
 }
 
+func (l javaLang) collectRuntimeDeps(kind, name string, file *rule.File) *sorted_set.SortedSet[label.Label] {
+	runtimeDeps := sorted_set.NewSortedSetFn([]label.Label{}, labelLess)
+	if file == nil {
+		return runtimeDeps
+	}
+
+	for _, r := range file.Rules {
+		if r.Kind() != kind || r.Name() != name {
+			continue
+		}
+
+		// This does not support non string list values from runtime_deps.
```

Comment:
```suggestion
		// This does not support non string list values from runtime_deps.
		// Currently, that means if a target has a runtime_deps of a different kind (e.g. a select), we will remove it.
		// Hopefully in the future we can be less destructive.
```

Reply:
Add in b040e7311788ff875faf389229864421af2f5843, with an example that shows this behavior.